pkg/middleware/recovery/recovery.go

```diff
@@ -0,0 +1,48 @@
+package recovery
+
+import (
+	"context"
+	"net/http"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func Panic(next http.Handler, logger logger.Logger) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				logger.Error("Recovered from panic",
+					zap.Any("panic_info", err))
+				http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
```

Comment:
Our API is `content/type: application/json`, so it's odd to me to be returning non-json body here.

Reply:
These are of type error

Reply:
Don't use `http.Error`. Favor being explicit in the content response:

```go
w.WriteHeader(http.StatusInternalServerError)
w.Header().Set("content-type", "application/json")

responseBody, err := json.Marshal(map[string]string{
    "code":    openfgav1.InternalErrorCode_internal_error,
    "message": errors.InternalServerErrorMsg,
})
if err != nil {
    logger.Error(
        "failed to JSON marshal HTTP response body",
        zap.Error(err),
    )
}

_, err = w.Write(responseBody)
if err != nil {
    logger.Error(
        "failed to write HTTP response body",
        zap.Error(err),
    )
}
```

Reply:
done